controller/resource_test.go

```diff
@@ -186,6 +186,144 @@ func TestTask_Upsert(t *testing.T) {
 	}
 }
 
-func TestTask_Delete(t *testing.T) {}
+func TestTask_Delete(t *testing.T) {
+	cases := map[string]struct {
+		createExistingToken                bool
+		updateExistingSecret               bool
+		task                               ecs.Task
+		expectTokenDeletedAndSecretEmptied bool
+	}{
+		"the token for service doesn't exist in consul and the secret is empty": {
+			createExistingToken:  false,
+			updateExistingSecret: false,
+			task: ecs.Task{
+				TaskArn:           pointerToStr("task"),
```

Comment:
nit: [`aws.String`](https://docs.aws.amazon.com/sdk-for-go/api/aws/#hdr-Value_and_Pointer_Conversion_Utilities) gives you a pointer

Reply:
yeah great call!